examples/gno.land/p/gov/dao/proposal.gno

```diff
@@ -0,0 +1,47 @@
+package dao
+
+import "std"
+
+type Status string
+
+// ACTIVE -> ACCEPTED -> EXECUTED
+// ACTIVE -> NOT ACCEPTED
+var (
+	Active      Status = "active"       // proposal is still active
+	Accepted    Status = "accepted"     // proposal gathered quorum
+	NotAccepted Status = "not accepted" // proposal failed to gather quorum
+	Executed    Status = "executed"     // proposal is executed
+)
+
+func (s Status) String() string {
+	return string(s)
+}
+
+// PropStore defines the proposal storage abstraction
+type PropStore interface {
+	// GetProposals returns the given paginated proposals
+	GetProposals(offset, count uint64) []Proposal
+
+	// GetProposalByID returns the proposal associated with
+	// the given ID, if any
+	GetProposalByID(id uint64) (Proposal, error)
+
+	// GetProposalsByAddress returns the proposals associated
+	// with the given proposer address
+	GetProposalsByAddress(address std.Address) []Proposal
+}
+
+// Proposal is the single proposal abstraction
+type Proposal interface {
+	// GetAuthor returns the author of the proposal
+	GetAuthor() std.Address
+
+	// GetDescription returns the description of the proposal
+	GetDescription() string
+
+	// GetStatus returns the status of the proposal
+	GetStatus() Status
+
+	// GetVotes returns the votes of the proposal
+	GetVotes() []Vote
```

Comment:
i think that this helper should not exists at all, in favor of indexers

Reply:
I get your point and I agree, but if we want to have "wrapper" realms that will serve some kind of UI-friendly API, we need export methods for data contained within the state, on-chain

There is no way to fetch this info on-chain, but through `GetVotes`

Reply:
Let’s do it with pagination and encourage people to focus on “recent” ones, also we need a filter to only have active ones VS archived ones

Reply:
Added pagination:
[ef299bd](https://github.com/gnolang/gno/pull/2581/commits/ef299bd6fa56833e5d9cf690d5daa5d5ac45e996)